native/more/relationship-list.react.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+
+import type { NavigationRoute } from '../navigation/route-names';
+import type { AppState } from '../redux/redux-setup';
+import type { MoreNavigationProp } from './more.react';
+
+import React from 'react';
+import { View, Text, FlatList } from 'react-native';
+
+import { connect } from 'lib/utils/redux-utils';
+import { type UserInfo } from 'lib/types/user-types';
+
+import { AddFriendsModalRouteName } from '../navigation/route-names';
+import { styleSelector } from '../themes/colors';
+
+import RelationshipListItem from './relationship-list-item.react';
+import RelationshipListAddButton from './relationship-list-add-button.react';
+
+type Props = {|
+  navigation: MoreNavigationProp<'FriendList'>,
+  route: NavigationRoute<'FriendList'>,
+  // Redux state
+  styles: typeof styles,
+|};
+
+type State = {||};
+
+const DATA = [
+  { id: '1', username: 'John' },
+  { id: '2', username: 'Emma' },
+];
+
+class RelationshipList extends React.PureComponent<Props, State> {
+  render() {
+    return (
+      <View style={this.props.styles.container}>
+        <FlatList
+          contentContainerStyle={this.props.styles.contentContainer}
+          ListHeaderComponent={
```

Comment:
I'm not sure, but I think right now this is forcing the `FlatList` to rerender everything every time `RelationshipList` is rendered. Besides that it feels a bit icky to have a huge prop like this defined inline.

Some thoughts:

1. Probably the easiest option for now would be to define a function component for this
2. But that wouldn't work well when `data` changes since you need access to that state. You could instead create a memoized render function that returns `ListHeaderComponent` that only changes when `DATA.length !== 0` changes
3. But on a higher level, when you find yourself defining a `ListHeaderComponent` that is actually dynamic, that should be a strong signal that this component should be defined as a normal `data` row
4. Even higher level: I'm not sure this is the best place to put the button for this. What if we placed it in `headerRight`, similar to where `ComposeThreadButton` is on `MessageList`?

Reply:
I agree that it would be better to keep it as a memoized function component. And you're 100% right, it would be much better to put it in the header. I will do that